### ISSUE TITLE
Version 0.9.3, lost rpc port when agent got a HUP SIG

### ISSUE THREAD

**Issue by HZ89** [victorcoder/dkron#262]
When the dkron agent got a HUP SIG, run
```
func (a *AgentCommand) handleReload() {
	a.Ui.Output("Reloading configuration...")
	newConf := ReadConfig(a)
	if newConf == nil {
		a.Ui.Error(fmt.Sprintf("Failed to reload configs"))
		return
	} else {
		a.config = newConf
	}
	// Reset serf tags
	a.serf.SetTags(a.config.Tags)
	//Config reloading will also reload Notification settings
}
```
reload configuriton,and advertise_rpc_port do not in the configurition file. the agent will lost rpc port

I add serf restart process in handleReload func, fix this problem
```
func (a *AgentCommand) handleReload() {
	a.Ui.Output("Reloading configuration...")
	newConf := ReadConfig(a)
	if newConf == nil {
		a.Ui.Error(fmt.Sprintf("Failed to reload configs"))
		return
	} else {
		a.config = newConf
	}

	//Reload serf
	if a.serf.State().String() != "shutdown" {
		a.Ui.Output("Gracefully shutting down serf")
		log.Infof("agent: Gracefully shutting down serf")
		if err := a.serf.Leave(); err != nil {
			a.Ui.Error(fmt.Sprintf("Error: %s", err))
			log.Error(fmt.Sprintf("Error: %s", err))
		}
		if err := a.serf.Shutdown(); err != nil {
			a.Ui.Error(fmt.Sprintf("Error: %s", err))
			log.Error(fmt.Sprintf("Errot: %s", err))
		}
		a.serf = nil
	}
	if a.serf = a.setupSerf(); a.serf == nil {
		log.Fatal("agent: Restart serf failured")
	}

	// Reset serf tags
	a.serf.SetTags(a.config.Tags)
	//Config reloading will also reload Notification settings

}
```

**Comment by victorcoder** [comment]
🤔 This probably works but I don't think restarting serf is a good thing to do regarding reliability.

This needs to be fixed in handleReload by fixing the advertise rpc port param.

**Comment by victorcoder** [comment]
Fixed in #360
